### DISCUSSION TITLE
Sender Id application form

### DISCUSSION THREAD

**Discussion by lgalloherran** [twilio-labs/paste#2936] [Office hours] [answered]
Hi Team! Happy new year! 
I want to review with you the newSender ID form and some components I expect to replicate for other forms. 

**1. Stepper**
<img width="256" alt="Captura de pantalla 2023-01-11 a la(s) 1 27 32 p m" src="https://user-images.githubusercontent.com/72419222/211888100-0f4226d5-a467-4b7d-9bf1-6a07b7fc61e1.png">

**2. Status Card:**
<img width="277" alt="Captura de pantalla 2023-01-11 a la(s) 1 28 12 p m" src="https://user-images.githubusercontent.com/72419222/211888240-3fcce255-0223-4d9f-a88e-a29aa6a5edb2.png">

**3. Actions bar**
<img width="878" alt="Captura de pantalla 2023-01-11 a la(s) 1 28 39 p m" src="https://user-images.githubusercontent.com/72419222/211888332-ab0081a5-6c9a-4088-a369-b2231c2a0f9b.png">

**4. Document dropper**
<img width="877" alt="Captura de pantalla 2023-01-11 a la(s) 1 28 56 p m" src="https://user-images.githubusercontent.com/72419222/211888397-0931fd60-1d88-4010-b7e2-49b0595a7ccf.png">

**5. Extra input upon radio or checklist selection:** This is not for this form yet but we expect it to be used soon. 
<img width="802" alt="Captura de pantalla 2023-01-11 a la(s) 1 29 47 p m" src="https://user-images.githubusercontent.com/72419222/211888576-56a4f984-5a97-46a8-9a3f-c2c00264ba74.png">

This is the[ link to the prototype](https://www.figma.com/proto/zKX4bWwAMbb6ReDSwym1cx/PN--Alphanumeric-Senders---Registration-Form-(2022)?page-id=107%3A150926&node-id=1143%3A254138&viewport=-4704%2C588%2C0.42&scaling=min-zoom&starting-point-node-id=1054%3A239601&show-proto-sidebar=1). 
Any other comments will be well received! 
Thanks for your time!

**Comment by nkrantz** [comment, marked as answer]
Hey @lgalloherran, thanks for coming to Office Hours yesterday.

[Here's the link to the recording](https://www.loom.com/share/956d9891613f4634af392f75f71dc478) (password in the #help-design-system slack channel).

1. [This is the Figma file of the Stepper design exploration work](https://www.figma.com/file/K1ftrUI2O7VYKZcKH2Ghdl/Stepper-component-brainstorming?node-id=142%3A22842&t=HNqoHtO7Ev9B7s3n-1 ) in case you'd like to reference it, but yours looks good.
2. For the card, we'd encourage you to think about whether the information needs to be outside of the main body of the page. Is it in the same position and does it have the same content throughout the entire flow? Be sure to think about what will happen with those side-by-side horizontal UIs on small screen sizes. You'll likely need to add a horizontal scroll bar.
3. Stick to the regular console usage for the actions bar.
4. Check out our new [File Uploader component](https://paste.twilio.design/components/file-uploader)! It's already in code, and the Figma assets will be available shortly ([here's the branch](https://www.figma.com/file/mMS8HcxvnDoN8ZeJxqaQey/feat(File-Uploader)%3A-add-component?node-id=23927%3A248210 )). We'd recommend removing the Cards wrapping your file uploaders - maybe use Separators instead, if that visual separation is important.
5. It's ok to dynamically show the input fields on checkbox selection, as long as the content that's appearing is below the user interaction on the page. Each input will also need its own associated Label - using placeholder text as a label can be confusing because it goes away as soon as there's any value in the input.

Let us know how your progress on this flow goes!
